PaintingTrivia.py

```diff
@@ -119,14 +128,49 @@ def show_entries():
             'selected_painter_id'] else 'wrong_guesses'
         session[key] += 1
 
-    painters = Painter.query.order_by(func.random()).limit(4).all()
+    while True:
+        painters = Painter.query.order_by(func.random()).limit(4).all()
+        selected_painter = random.choice(painters)
+        selected_painting = Painting.query.filter(Painting.painter == selected_painter).order_by(func.random()).limit(
+            1).first()
+        if selected_painting:
+            break
```

Comment:
smth along these lines:
http://stackoverflow.com/questions/6044309/sqlalchemy-how-to-join-several-tables-by-one-query

On Wed, Jan 13, 2016 at 11:21 AM, Jeronics <notifications@github.com> wrote:

> In PaintingTrivia.py
> <https://github.com/denisergashbaev/PaintingTrivia/pull/3#discussion_r49572276>
> :
>
> > @@ -119,14 +128,49 @@ def show_entries():
> >              'selected_painter_id'] else 'wrong_guesses'
> >          session[key] += 1
> >
> > -    painters = Painter.query.order_by(func.random()).limit(4).all()
> > +    while True:
> > +        painters = Painter.query.order_by(func.random()).limit(4).all()
> > +        selected_painter = random.choice(painters)
> > +        selected_painting = Painting.query.filter(Painting.painter == selected_painter).order_by(func.random()).limit(
> > +            1).first()
> > +        if selected_painting:
> > +            break
>
> Yes, I am thinking about how to do it.. this was kind of a quick fix,
> we'll discuss it tonight!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/denisergashbaev/PaintingTrivia/pull/3/files#r49572276>
> .
>